pkg/containerutil/cp_linux.go

```diff
@@ -247,3 +282,50 @@ func mountSnapshotForContainer(ctx context.Context, client *containerd.Client, c
 	}
 	return tempDir, cleanup, nil
 }
+
+func getContainerMountInfo(ctx context.Context, con containerd.Container, containerPath string, container2host bool) (string, string, error) {
+	filePath := filepath.Clean(containerPath)
+	spec, err := con.Spec(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	// read-only applies only while copying into container from host
+	if !container2host && spec.Root.Readonly {
+		return "", "", fmt.Errorf("container rootfs: %s is marked read-only", spec.Root.Path)
+	}
+
+	for _, mount := range spec.Mounts {
```

Comment:
It seems that the `nerdctl cp` behaviour is different between running and stopping containers when they have duplicated volumes:

When a file is copied to the runing container, it looks file the file is copied to the second volue:

```console
# nerdctl run -it --name voltest -v test1:/mnt/ -v test2:/mnt/ ubuntu:22.04 /bin/bash
...
# echo hi > /tmp/hi
# nerdctl cp /tmp/hi voltest:/mnt/
# nerdctl run --rm -it -v test2:/mnt/ ubuntu:22.04 cat /mnt/hi
hi
```

After stopping `voltest`:

```console
# echo aaa > /tmp/hi2
# nerdctl cp /tmp/hi2 voltest:/mnt/
# nerdctl run --rm -it -v test1:/mnt/ ubuntu:22.04 ls -al /mnt/
total 12
drwxr-xr-x 2 root root 4096 Aug 26 06:25 .
drwxr-xr-x 1 root root 4096 Aug 26 06:27 ..
-rw-r--r-- 1 root root    4 Aug 26 06:25 mnt
# nerdctl run --rm -it -v test1:/mnt/ ubuntu:22.04 cat /mnt/mnt
aaa
```

Now it looks like the file `hi2` is copied to the volume `test1` and the name also changed to `mnt` (not `hi2`).

Reply:
@ktock This is a valid point, however having duplicate mount points does not make a lot of sense IMO. Maybe this fix should be in `run_mount.go` to not allow duplicate mount points and keep the behaviour consistent with docker? Test in docker. 
```
 docker run -it --name voltest -v test1:/mnt/ -v test2:/mnt/ ubuntu:22.04 /bin/bash
Unable to find image 'ubuntu:22.04' locally
22.04: Pulling from library/ubuntu
db76c1f8aa17: Pull complete 
Digest: sha256:ec050c32e4a6085b423d36ecd025c0d3ff00c38ab93a3d71a460ff1c44fa6d77
Status: Downloaded newer image for ubuntu:22.04
docker: Error response from daemon: Duplicate mount point: /mnt.
See 'docker run --help'.
```

Reply:
> not allow duplicate mount points and keep the behaviour consistent with docker

SGTM